Examples/Google/NaturalLanguageNIO/Sources/main.swift

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Dispatch
+import Foundation
+import SwiftGRPCNIO
+import OAuth2
+import NIO
+import NIOHTTP1
+import NIOSSL
+
+func makeClientTLS() throws -> GRPCClientConnection.TLSMode {
+  let configuration = TLSConfiguration.forClient(applicationProtocols: ["h2"])
+  let context = try NIOSSLContext(configuration: configuration)
+  return .custom(context)
+}
+
+let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+
+/// Create a client and wait for it to initialize. Returns nil if initialisation fails.
+func makeServiceClient(address: String, port: Int) -> Google_Cloud_Language_V1_LanguageServiceService_NIOClient? {
+  do {
+    return try GRPCClientConnection.start(host: address,
+                                          port: port,
+                                          eventLoopGroup: eventLoopGroup,
+                                          tls: makeClientTLS())
+      .map { client in
+        Google_Cloud_Language_V1_LanguageServiceService_NIOClient(connection: client)
+      }
+      .wait()
+  } catch {
+    print("Unable to create a client: \(error)")
+    return nil
+  }
+}
+
+enum AuthError: Error {
+  case providerFailed
+  case noProvider
+}
+
+func getAuthToken(_ eventLoopGroup: MultiThreadedEventLoopGroup,
```

Comment:
How about passing in an `eventLoop` instead?

Reply:
done